nessie/lakectl_test.go

```diff
@@ -209,3 +209,72 @@ func TestLakectlMergeAndStrategies(t *testing.T) {
 	RunCmdAndVerifySuccessWithFile(t, Lakectl()+" merge lakefs://"+repoName+"/"+mainBranch+" lakefs://"+repoName+"/"+featureBranch+" --strategy dest-wins", false, "lakectl_merge_source_wins", branchVars)
 	RunCmdAndVerifySuccessWithFile(t, Lakectl()+" fs ls lakefs://"+repoName+"/"+featureBranch+"/", false, "lakectl_fs_ls_1_file", vars)
 }
+
+func TestLakectlAnnotate(t *testing.T) {
+	repoName := generateUniqueRepositoryName()
+	storage := generateUniqueStorageNamespace(repoName)
+	vars := map[string]string{
+		"REPO":    repoName,
+		"STORAGE": storage,
+		"BRANCH":  mainBranch,
+	}
+
+	// create fresh repo with 'main' branch
+	RunCmdAndVerifySuccessWithFile(t, Lakectl()+" repo create lakefs://"+repoName+" "+storage, false, "lakectl_repo_create", vars)
```

Comment:
Suggestion to use the table driven format - it will structure the following code and will report each one as a test:

```go
	tests := []struct {
		Name   string
		Cmd    string
		Golden string
		Vars   map[string]string
	}{
		{
			Name:   "log not found",
			Cmd:    "log lakefs://" + repoName + "/" + mainBranch,
			Golden: "lakectl_log_404",
			Vars:   map[string]string{"REPO": repoName, "STORAGE": storage, "BRANCH": mainBranch},
		},
	}
	for _, tt := range tests {
		t.Run(tt.Name, func(t *testing.T) {
			RunCmdAndVerifyFailureWithFile(t, Lakectl()+" "+tt.Cmd, false, tt.Golden, vars)
		})
	}
```

Reply:
@nopcoder - I like the idea, but I think it is too large of a change to be included with this bug fix.
How about I create an issue for that?

Reply:
Opened #3021 to handle this